### ISSUE TITLE
Error when serializing data from module SMS: Circular reference detected

### ISSUE THREAD

**Issue by Shellcat-Zero** [mvt-project/mvt#443]
When running:
`mvt-ios check-backup --output ./check_backup decrypted_backup/`

I am getting the error:
```
14:27:50 ERROR    [mvt.ios.modules.mixed.sms] Error when serializing data from module SMS: Circular reference detected                                                                                                                                                         
                  Traceback (most recent call last):                                                                                                                                                                                                                           
                    File "/home/user/miniconda3/lib/python3.11/site-packages/mvt/common/module.py", line 217, in run_module                                                                                                                                                      
                      module.to_timeline()                                                                                                                                                                                                                                     
                    File "/home/user/miniconda3/lib/python3.11/site-packages/mvt/common/module.py", line 155, in to_timeline                                                                                                                                                     
                      self.timeline = self._deduplicate_timeline(self.timeline)                                                                                                                                                                                                
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                
                    File "/home/user/miniconda3/lib/python3.11/site-packages/mvt/common/module.py", line 133, in _deduplicate_timeline                                                                                                                                           
                      timeline_set.add(json.dumps(record, sort_keys=True))                                                                                                                                                                                                     
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                      
                    File "/home/user/miniconda3/lib/python3.11/json/__init__.py", line 238, in dumps                                                                                                                                                                             
                      **kw).encode(obj)                                                                                                                                                                                                                                        
                            ^^^^^^^^^^^                                                                                                                                                                                                                                        
                    File "/home/user/miniconda3/lib/python3.11/json/encoder.py", line 200, in encode                                                                                                                                                                             
                      chunks = self.iterencode(o, _one_shot=True)                                                                                                                                                                                                              
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                              
                    File "/home/user/miniconda3/lib/python3.11/json/encoder.py", line 258, in iterencode                                                                                                                                                                         
                      return _iterencode(o, 0)                                                                                                                                                                                                                                 
                             ^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                                 
                  ValueError: Circular reference detected
```
I'm running python 3.11.5 and MVT==2.4.5 after cloning the repo. I encountered this error after pulling in the `uri=True` fix from #434.

**Comment by roaree** [comment]
@Shellcat-Zero Can you confirm if the same error occurs with `mvt==2.4.3` from PyPI? It looks like another bug, but I would be quite surprised if the change introduced in #442 caused this.

**Comment by roaree** [comment]
SMS module had recent changes in #438 that are not yet in a PyPI release but are in `main` - seems related.

**Comment by roaree** [comment]
Thanks for the report! Feel free to reopen if the fix in #444 didn't fix this bug

**Comment by Shellcat-Zero** [comment]
@roaree 2.4.3 from PyPI does not produce the error. After pulling the latest commits, it appears 2.5.0 works now without the error. Thanks!
